### PR TITLE
hopefully complete metadata checks:

### PR DESCRIPTION
**Pull request by scseeman**
- dissection time
- acsf
- internal
- project code
- target region
- rig name

- [ ] TODO: implement flag to actually set parameters

### CODE REVIEW

File: util/metadata_cleanup.py [hunk 1]
```diff
@@ -105,17 +106,115 @@ def dissection_check(path, dh, sub_id, expt_date, columns):
...
+        else:
+            print("\tTarget region mismatch: %s != V1" % region)
+
+def internal_check(dh, species):
+    internal = dh.info().get('internal', None)
+    internal_dye = dh.info().get('internal_dye', None)
+    if internal is None:
+        print("\tSet Internal: Standard K-Gluc")
+        return
+    if internal.lower() != 'standard k-gluc':
```
Comment (campagnola) [inline util/metadata_cleanup.py]: check if it's in the list of known inernals (in mp_a.constants)

File: util/metadata_cleanup.py [hunk 2]
```diff
@@ -105,17 +106,115 @@ def dissection_check(path, dh, sub_id, expt_date, columns):
...
+        print("\tInternal mismatch: %s != Standard K-Gluc" % internal)
+        return
+    if internal_dye is not None:
```
Comment (campagnola) [inline util/metadata_cleanup.py]: doesn't get checked if we hit a return above

Comment (campagnola) [reply util/metadata_cleanup.py]: (maybe just split into two functions)

File: util/metadata_cleanup.py [hunk 3]
```diff
@@ -105,17 +106,115 @@ def dissection_check(path, dh, sub_id, expt_date, columns):
...
+    if internal_dye is None and species != 'mouse':
+        genotype = species.split(';')
+        if len(genotype) < 3:
+            print("\tSet internal dye: AF488, %s looks likes single transgenic" % species)
+        else:
+            print("\tSet internal dye: Cascade Blue, %s looks liked quad" % species)
+    else:
+        print("\tCan't parse genotype %s, set internal dye manually" % species)
+
+def rig_check(dh):
```
Comment (campagnola) [inline util/metadata_cleanup.py]: I'm confused about the logic here.. maybe:

    def rig_check(dh):
        rig = dh.info().get('rig_name', None)
        if rig in (None, ''):
            #dh.setInfo(rig_name=config.rig_name)
            print("\tRig set: %s" % config.rig_name)
            return
        elif rig != config.rig_name:
            print("\tRig mismatch: %s != %s" % (rig, config.rig_name))

File: util/metadata_cleanup.py [hunk 4]
```diff
@@ -105,17 +106,115 @@ def dissection_check(path, dh, sub_id, expt_date, columns):
...
+        if species is None and sub_id is None:
+            print("\tNo specimen, can't set project code")
+            return
+        if species.lower() == 'human':
+            print("\tSet Project Code: human course matrix")
+            return
+        if expt_date < mouse_prod:
+            print("\tSet Project Code: mouse V1 pre-production")
+        else:
+            print("\tSet Project Code: mouse V1 course matrix")
```
Comment (campagnola) [inline util/metadata_cleanup.py]: coarse

File: util/metadata_cleanup.py [hunk 5]
```diff
@@ -105,17 +106,115 @@ def dissection_check(path, dh, sub_id, expt_date, columns):
...
 
+def project_check(dh, sub_id, expt_date, species):
+    mouse_prod = datetime.date(2017, 10, 01)
+    project = dh.info().get('project', None)
+    if project is None:
+        if species is None and sub_id is None:
+            print("\tNo specimen, can't set project code")
+            return
+        if species.lower() == 'human':
+            print("\tSet Project Code: human course matrix")
```
Comment (campagnola) [inline util/metadata_cleanup.py]: coarse

File: util/metadata_cleanup.py (hunk 3 above)
Comment (scseeman) [reply util/metadata_cleanup.py]: do you want the `elif rig != config.rig_name` just to be explicit? as opposed to just leaving it as an `else`, which would also catch any unforeseen weirdness

File: util/metadata_cleanup.py [hunk 6]
```diff
@@ -101,21 +99,107 @@ def dissection_check(path, dh, sub_id, expt_date, columns):
...
+
+    if internal_dye is None and species is None:
+        print("\tCan't set internal dye")
+        return
+    if internal_dye is None and species.lower() == 'human':
+        print("\tSet internal dye: AF488")
+        return
+    if internal_dye is None and genotype is not None:
+        if len(genotype.split(';')) < 3:
+            print("\tSet internal dye: AF488, %s looks likes single transgenic" % genotype)
```
Comment (campagnola) [inline util/metadata_cleanup.py]: Could this give the wrong answer for failed quads with GFP? Another option is:
```
dye = 'Cascade Blue' if 'green' in Genotype(genotype).colors else 'AF488'
```

File: util/metadata_cleanup.py (hunk 3 above)
Comment (campagnola) [reply util/metadata_cleanup.py]: I think the only other possibility would be `rig == config.rig_name`, which can (and should) be safely ignored.

**Pull request merged**